File: LeetCode1/1.两数之和.py
# @author: PiggerZZM
# @lc app=leetcode.cn id=1 lang=python3
# @date: 2020-6-22
# [1] 两数之和
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        hash_map = {}
        for index, value in enumerate(nums):
            hash_map[value] = index
        for index, value in enumerate(nums):
            if target - value in hash_map and hash_map[target - value] != index:
                result.append(index)
                result.append(hash_map[target - value])
                break

        return result
# @lc code=end

File: LeetCode1/1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 * author: PiggerZZM
 * date: 2020-6-22
 * [1] 两数之和
 */

// @lc code=start
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        map<int, int> hash;
        vector<int> result;
        hash.clear();
        for (int i = 0; i < nums.size(); i++)
            hash[nums[i]] = i;
        for (int i = 0; i < nums.size(); i++)
        {
            map<int, int>::iterator it = hash.find(target - nums[i]);
            if (it != hash.end() && i != it->second)
            {
                result.push_back(i);
                result.push_back(it->second);
                break;
            }
        }
        return result;
    }
};
// @lc code=end
